account_invoice_merge/invoice.py

```diff
@@ -207,8 +206,9 @@ def make_key(br, fields):
                             [('product_id', '=', so_line.product_id.id),
                              ('invoice_id', '=', new_invoice_id)])
                         if invoice_line_ids:
+                            invoice_line_ids_list = [x.id for x in invoice_line_ids]
```

Comment:
Would be better if you renamed that variable `invoice_lines` as search returns a record set. Thus you can do: `invoice_lines.ids` to get a list of ids.